app/src/main/assets/GimelGimel/CesiumEventHandler.js

```diff
@@ -48,43 +49,60 @@ GG.EventHandler.prototype.setScreenSpaceEventAction = function (screenSpaceEvent
 };
 
 
-GG.EventHandler.prototype.setSingleTouchActions = function (layersManager) {
-
-    var handler = this;
-    //Set mouse move event listener
-    //Cesium.ScreenSpaceEventType.LEFT_DOWN is mapped to touch events.
-    this.setScreenSpaceEventAction(Cesium.ScreenSpaceEventType.LEFT_DOWN, function (movement) {
-        var entity = handler.pickEntity(movement.position);
-        clickEntityWithLayer(entity);
-
-        var cartesian = handler._viewer.camera.pickEllipsoid(movement.position, handler._scene.globe.ellipsoid);
-        if (cartesian) {
-            var cartographic = Cesium.Cartographic.fromCartesian(cartesian);
-            var longitude = Cesium.Math.toDegrees(cartographic.longitude);
-            var latitude = Cesium.Math.toDegrees(cartographic.latitude);
-
-            GG.AndroidAPI.updateSelectedLocation({
-                            longitude: longitude,
-                            latitude: latitude
-                        });
-
-
-        }
-    });
+GG.EventHandler.prototype.onSingleTap = function (relativeX, relativeY) {
+    var self = this;
+    var position = self.getWindowPosition(relativeX, relativeY);
+    var entity = this.pickEntity(position);
+    clickEntityWithLayer(entity);
 
     function clickEntityWithLayer(entity) {
```

Comment:
I'm not sure if you did it, but function inside a function isn't good.

It should be split to a separate "standalone" function.